.github/workflows/build-binary.yml

```diff
@@ -0,0 +1,122 @@
+name: Build java app image
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+    inputs:
+      sem-version:
+        description: 'Version'
+        required: false
+
+env:
+  JAVA_DIST: 'zulu'
+  JAVA_VERSION: '22.0.2+9'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  prepare:
+    name: Determines the versions strings for the binaries
+    runs-on: [ubuntu-latest]
+    outputs:
+      semVerStr: ${{ steps.determine-version.outputs.version }}
+      semVerNum: ${{steps.determine-number.outputs.number}}
+    steps:
+      - id: determine-version
+        shell: pwsh
+        run: |
+          if ( ${{github.event_name}} -eq 'release') {
+            echo "version=${{ github.event.release.tag_name}}" >> $GITHUB_OUTPUT
+          } else if (${{inputs.sem-version}}) {
+             echo "version=${{ inputs.sem-version}}" >> $GITHUB_OUTPUT
+          }
+          Write-Error "Version neither via input nor by tag specified. Aborting"
+          exit 1
+      - id: determine-number
+        run: |
+          SEM_VER_NUM=`echo ${{ steps.determine-version.outputs.version }} | sed -E 's/([0-9]+\.[0-9]+\.[0-9]+).*/\1/'`
+          echo "number=${SEM_VER_NUM}" >> $GITHUB_OUTPUT
+
+  build-win:
+    name: Build java app image on windows
+    needs: [prepare]
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            architecture: x64
+            native-access-lib: 'org.cryptomator.jfuse.linux.amd64'
+          - os: [self-hosted, Linux, ARM64]
+            architecture: aarch64
+            native-access-lib: 'org.cryptomator.jfuse.linux.aarch64'
+          - os: [self-hosted, macOS, ARM64]
+            architecture: aarch64
+            native-access-lib: 'org.cryptomator.jfuse.mac'
+          - os: macos-latest
+            architecture: x64
+            native-access-lib: 'org.cryptomator.jfuse.mac'
+          - os: windows-latest
+            architecture: x64
+            native-access-lib: 'org.cryptomator.jfuse.win'
+    runs-on: ${{ matrix.os }}
```

Comment:
@coderabbitai your second point does not make sense. Our matrix strategy only specifies certain instances, since we leave out arrays to combine with each other and only use the `include` list.